src/DynamoCoreWpf/ViewModels/Core/NodeViewModel.cs

```diff
@@ -397,6 +500,22 @@ public NodeViewModel(WorkspaceViewModel workspaceViewModel, NodeModel logic)
             ShowExecutionPreview = workspaceViewModel.DynamoViewModel.ShowRunPreview;
             IsNodeAddedRecently = true;
             DynamoSelection.Instance.Selection.CollectionChanged += SelectionOnCollectionChanged;
+            WorkspaceViewModel.PropertyChanged +=WorkspaceViewModel_PropertyChanged;
+        }
+
+        private void WorkspaceViewModel_PropertyChanged(object sender, PropertyChangedEventArgs e)
+        {
+            switch (e.PropertyName)
```

Comment:
I don't think the node needs to be responding to workspace property changes. Just have the workspace view model set the nodes values to what they need to be.